discord/state.py

```diff
@@ -967,8 +967,19 @@ def parse_guild_role_update(self, data):
     def parse_guild_members_chunk(self, data):
         guild_id = int(data['guild_id'])
         guild = self._get_guild(guild_id)
+        presences = data.get('presences', [])
+
         members = [Member(guild=guild, data=member, state=self) for member in data.get('members', [])]
         log.debug('Processed a chunk for %s members in guild ID %s.', len(members), guild_id)
+
+        if presences:
+            member_dict = {str(member.id): member for member in members}
+            for presence in presences:
+                user = presence['user']
+                member_id = user['id']
+                member = member_dict.get(member_id)
+                member._presence_update(presence, user)
```

Comment:
Since the `presences` key will mostly be missing, use a `try except KeyError` instead. 

Might be worth swapping the lookup table (`{user_id: presence}` instead of `{user_id: member}`) though this point is minor.